Modules/Report/Services/Finance/RevenueReportService.php

```diff
@@ -59,17 +60,29 @@ private function getParticularAmountForExport(array $particular, Object $startDa
         $invoices = $this->invoiceService->getInvoicesBetweenDates($startDate, $endDate, 'non-indian');
         $totalAmount = 0;
         $results = [];
+        $currency_chages = [];
 
         // ToDo:: We need to change this logic and get the exchange rate for every month.
-        $exchangeRates = app(CurrencyService::class)->getCurrentRatesInINR();
-
+        $exchangeRate = [];
+        $exchangeRates = CurrencyAvgRate::select('avg_rate', 'captured_for')->groupBy('captured_for')->get()->toArray();
+        $exchangeMonth = null;
+        foreach ($exchangeRates as $exchangeRate) {
+            $exchangeMonth = (date('m-y', strtotime($exchangeRate['captured_for'])));
+            $currency_chages[] = [
+                $exchangeMonth => $exchangeRate['avg_rate']
+            ];
+        }
         foreach ($invoices as $invoice) {
-            $amount = $invoice->amount * $exchangeRates;
             $dateKey = $invoice->sent_on->format('m-y');
+            if ($exchangeMonth == $dateKey) {
+                $exchangeDollor = $exchangeRate['avg_rate'];
```

Comment:
I think this should be like this right ?

```suggestion
            if (isset($currency_chages[$dateKey])) {
                $exchangeDollor = $currency_chages[$dateKey];
```

Reply:
Please test by checking if the invoice amount is calculating based on the other the currency value. 

Like if a April invoice amount is 100
and April avg rate is 70 

Then the invoice value will be 7000

Reply:
$exchangeDollor = $currency_chages[$dateKey];     by doing this we will get only the month and year not the avg_rate

Reply:
So please why this is not working. 

Also check the exchangeRates array format. It should be
[
"month-Year" => "AvgRates"
]